### PR TITLE
Adding fix to handle corner case scenarios for eml files where boundary is not present but content is present.

### PR DESCRIPTION
**Pull request by reeshabh90**
Also added the test file in tests folder for verirfication

## Summary by CodeRabbit

* **Bug Fixes**
  * Improved email parsing for malformed multipart messages that contain text but no detectable parts.
  * Plain-text content is now extracted correctly instead of being omitted.
  * Added clearer error reporting when malformed multipart content cannot be processed.

* **Tests**
  * Added coverage for multipart emails missing boundary delimiters.

### CODE REVIEW

File: src/eml_parser.cpp [hunk 1]
```diff
@@ -238,8 +238,22 @@ struct pimpl_impl<eml_parser> : pimpl_impl_base
...
 		}
-		// TO DO: Handle the case, where boundary is declared but body does not
-		// contain a single boundary line.
+		else if (ct.media_type() == mime::media_type_t::MULTIPART && mime_entity.parts().empty() && !mime_entity.content().empty())
+		{
+			// Fix for Degenerate multipart: the header declared a multipart Content-Type,
+			// but the body never contained any boundary-delimited parts.
+			log_scope();
+			std::string plain_text = mime_entity.content();
+			plain_text.erase(std::remove(plain_text.begin(), plain_text.end(), '\r'), plain_text.end());
```
Comment (as-ascii) [inline src/eml_parser.cpp]: Is this required for sure or will be handled by plain text parser later? I think we have this normalization and more somewhere in the code already.

Comment (reeshabh90) [reply src/eml_parser.cpp]: You are correct, we have a normalization function in this file as well as txt_exporter.cpp also normalizes.

**Comment by as-ascii** [comment]
@coderabbitai full review

File: src/eml_parser.cpp [hunk 2]
```diff
@@ -161,6 +161,24 @@ struct pimpl_impl<eml_parser> : pimpl_impl_base
 				}
 			}
 		}
+		else if (ct.media_type() == mime::media_type_t::MULTIPART && mime_entity.parts().empty())
+		{
+			std::string plain_text = mime_entity.content();
+			if (!plain_text.empty())
```
[the remark this answers is not in the document]
Comment (as-ascii) [reply src/eml_parser.cpp:167]: Please double check if plain_text_extractor class does it. It does not make sense to duplicate the feature in my opinion if handling of contents is delegated to that class.

**Review by as-ascii — APPROVED** [review]
